Dockerfile

```diff
@@ -1,28 +1,32 @@
-FROM node:8
+FROM golang:alpine as golang
+RUN apk add --no-cache build-base git && \
+    go get github.com/tj/node-prune/cmd/node-prune
 
-ENV HOST localhost
-ENV PORT 3000
-
-# Create app directory
-RUN mkdir -p /usr/src/app
+FROM node:8-alpine as build
 WORKDIR /usr/src/app
-
-# Install GYP dependencies globally, will be used to code build other dependencies
-RUN npm install -g --production node-gyp && \
-    npm cache clean --force
-
-# Install app dependencies
-COPY package.json /usr/src/app
-RUN npm install --production && \
+COPY package*.json ./
+COPY --from=golang /go/bin/node-prune /usr/local/bin/
+RUN set -x && \
+    apk add --no-cache python build-base && \
+    npm install -g --production node-gyp && \
+    npm install --production && \
     npm install --production redis@0.10.0 talib@1.0.2 tulind@0.8.7 pg && \
```

Comment:
What's the reason for this not being in package.json?

Reply:
node-gyp has to be installed globally to build properly, talib and tulind are optional modules so they are not part of the package.json